packages/backend/middlewares/auth.js

```diff
@@ -0,0 +1,67 @@
+var jwt = require("jsonwebtoken");
+const serviceAccount = require("../firebaseServiceAccountKey.json");
+
+// TODO PR COMMENT Not sure why, but the custom error messages don't seem to be sent
+/**
+ * Middleware to validate any request that needs to be authorized.
+ *
+ * For the verification to work properly, two headers have to be sent with the request:
+ * ```js
+ * headers: {
+ *   authorization: `token ${token}`, // token being the one generated at `POST /sign`
+ *   address, // address being the blockchain address of the user making the request
+ * }
+ * ```
+ *
+ * See /packages/react-app/views/JwtTest.jsx for a code example.
+ * Testing of the middleware can be done at /jwt-test on the client side
+ *
+ * @param {Express.Request} req
+ * @param {Express.Response} res
+ * @param {Express.NextFunction} next
+ */
+const jwtAuth = (req, res, next) => {
+  const tokenHeader = req.headers.authorization;
+  const addressHeader = req.headers.address;
+  if (!tokenHeader) {
+    console.log("returning 401, no authorization header");
+    res.status(401).send("You need to sign to access this resource");
```

Comment:
Not sure why, but the message sent with `.send(message)` are not received on the client side. I've looked around for a bit, but didn't found anything useful.

If you know what's going on, we can fix it. Otherwise, I can just take the `.send()` part out, and we'll still have logs of what caused the 401 on the backend.

Reply:
I removed the messages and replaced `.status` with `.sendStatus`